indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -433,6 +438,76 @@ public static Path makeSegmentOutputPath(
     return outputPath;
   }
 
+  /**
+   * Rename the files. This works around some limitations of both FileContext (no s3n support) and NativeS3FileSystem.rename
+   * which will not overwrite
+   * @param outputFS The output fs
+   * @param indexZipFilePath The original file path
+   * @param finalIndexZipFilePath The to rename the original file to
+   * @return False if a rename failed, true otherwise (rename success or no rename needed)
+   * @throws IOException Bubbles up IOException from the hadoop classes.
+   */
+  private static boolean renameIndexFiles(
```

Comment:
minor thing, If these are just static util methods can they be moved to JobHelper, so that they can be reused by other jobs when required ?